src/components/ObjectInfo.ts

```diff
@@ -40,6 +42,24 @@ export default class ObjectInfo extends mixins(MixinUtil) {
       this.data = this.obj;
   }
 
+  async get_meta() {
+    if (!this.xmeta) {
+      const rname = this.getRankedUpActorNameForObj(this.data);
+      this.xmeta = await MsgMgr.getInstance().getObjectMetaData(rname);
+    }
+  }
+
+  private meta(item: string) {
+    if (!this.xmeta) {
+      this.get_meta();
+    }
+    if (this.xmeta) {
```

Comment:
These 4 lines can be replaced with `return this.xmeta?[item];` (might be worth adding a comment to explain that xmeta can still be null despite the null check in loadMetaIfNeeded, because we might be missing metadata for the object)